lms/views/email.py

```diff
@@ -76,12 +76,16 @@ def preferences_redirect(self):
         renderer="lms:templates/email/preferences.html.jinja2",
     )
     def preferences(self):
+        results = self.request.session.pop_flash("email_preferences_result")
```

Comment:
The `request` object is available within the Jinja2 template file (`preferences.html.jinja2`) so it's not actually necessary for the Python code to read the flash message and pass it to the template. Something like this:

```jinja2
{% for message in request.session.pop_flash("email_preferences_result") %}
  <p>{{ message }}</p>
{% endfor %}
```

You can also use `{% if request.session.peek_flash("email_preferences_result") %}` if you want to test whether there are any flash messages.

Reply:
I know, but in this case the template is mostly just rendering a `<script />` with JSON on it.

Since we need to pass the toast message result as part of that JSON, it was more convenient to read it here, rather than in the template itself.

See https://github.com/hypothesis/lms/blob/fdd3b83fd73a6aba0eb23d006da5bfc8eccab40c/lms/templates/email/preferences.html.jinja2#L16

Reply:
Ok, that's fine :+1: